PowerEditor/src/WinControls/ProjectPanel/TreeView.cpp

```diff
@@ -518,15 +550,16 @@ bool TreeView::searchLeafRecusivelyAndBuildTree(HTREEITEM tree2Build, const gene
 		size_t res = itemNameUpperCase.find(text2SearchUpperCase);
 		if (res != generic_string::npos)
 		{
-			if (tvItem.lParam)
-			{
-				tvItem.lParam = (LPARAM)(new generic_string(*((generic_string *)(tvItem.lParam))));
-			}
+			tvItem.lParam = (LPARAM) ((TreeViewData*) tvItem.lParam)->clone();
+
 			TVINSERTSTRUCT tvInsertStruct;
 			tvInsertStruct.item = tvItem;
 			tvInsertStruct.hInsertAfter = (HTREEITEM)TVI_LAST;
 			tvInsertStruct.hParent = tree2Build;
-			::SendMessage(_hSelf, TVM_INSERTITEM, 0, (LPARAM)(LPTVINSERTSTRUCT)&tvInsertStruct);
+			HTREEITEM hInsertedItem = (HTREEITEM)::SendMessage(_hSelf, TVM_INSERTITEM, 0, (LPARAM)(LPTVINSERTSTRUCT)&tvInsertStruct);

```

Comment:
You ***absolutely*** should check to see if `::SendMessage` returned `NULL` before using `newItem`.

***Especially*** since you're putting it in a variable named `_validHandles`.